### ISSUE TITLE
Esc key doesn't register, commands entered into code, random repositioning

### ISSUE THREAD

**Issue by jorhett** [vscode-neovim/vscode-neovim#826]
Even when Vscode is the only thing running, and there's over 48 GB of free memory and 8 cores available, vscode-neovim is super slow in responding to the esc key press. I've been using Vi for 35 years, and my fingers spin off `ESC :w` or `ESC b b` to go backwards or any other sequence really fast. Yet the behavior I've seen is as follows:

* The first 4-5 characters I type after hitting escape end up being added to the file
* Often the cursor moves to the beginning of the following line after processing the escape, so sequences like `ESC 6 d w` end up deleting words on entirely the wrong line

While VsCodeVim has many flaws, it doesn't suffer from a failure to respond to ESC keypresses in a timely manner. After nearly 4 decades using Vi I cannot retrain my fingers to stop typing for 2 seconds after hitting ESC, and the random repositions make commands risky.  I've learn far more about VsCode's ability to compare and revert changes than I ever wanted to know.

Points of relevance:
* I have no custom insert mappings, and I'm currently using a completely empty `init.vim`
* Neither nvim nor vimr have any problem with this
* VsCodeVim had tons of problems but never with accepting the escape sequence

**Comment by trkoch** [comment]
I've never experienced this. How to reproduce? Did you try with a vanilla VSCode config, in particular empty custom key bindings?

**Comment by theol0403** [comment]
Can't reproduce either.

**Comment by jorhett** [comment]
> I've never experienced this. How to reproduce? Did you try with a vanilla VSCode config, in particular empty custom key bindings? Do you see a message like "Waiting for second key of chord"?

I'm not sure why you asked this, as I said in my original report
> I have no custom insert mappings, and I'm currently using a completely empty init.vim

> How to reproduce?

Hit ESC-3b4dw and watch some of those keys end up on your screen.

It's 100% reproducible on every Mac I use. It's reproducable on a brand new mac that I've only loaded up Homebrew (to get neovim) and VScode on, zero other customization.  That makes it very obvious that the problem is within the plugin itself.

Disabling this plugin and enabling VscodeVim solves the delay/failure to track the state switch completely, albeit with the limited features that it provides.

> Can't reproduce either.
> theol0403 closed this as completed

I find this response very odd, given that 

* other people have reported this... 
  * https://github.com/vscode-neovim/vscode-neovim/issues/460
  * https://github.com/vscode-neovim/vscode-neovim/issues/495

* you have a PR to address typing lag https://github.com/vscode-neovim/vscode-neovim/pull/948  

* there's a fork of this project where the original change was to lower the time period it waits after escape is pressed -- which does limit but does not solve the problem https://marketplace.visualstudio.com/items?itemName=andyyu2004.vscode-neovim

In short, numerous people have reported this to be true. What is it that you need for proof?  Want a video of it happening?  I can easily provide, with system resources monitors on the side showing my M1 Pro Max macbook is otherwise bored

**Comment by theol0403** [comment]
@jorhett 
I closed this issue because we cannot reproduce, and there is no further information. It's possible it's a mac or homebrew issue, but I don't own one.

> I'm not sure why you asked this, as I said in my original report

you did not say you had empty `settings.json`, `keybindings.json`, or no vscode extensions (most likely culprit)

every issue/pr you linked to have nothing to do with this issue, except for vaguely #495. The rest (and the fork) refer to composite escape (jk), and my PR *adds* typing lag, not removes it (although it might as a side effect resolve any issue you were having).

**Comment by jorhett** [comment]
> you did not say you had empty `settings.json`, `keybindings.json`, or no vscode extensions (most likely culprit)

I don't know what the first two are. I have no modified json files that I'm aware of. I did say that I can reproduce on a fresh mac with zero other things installed. 

> It's reproducable on a brand new mac that I've only loaded up Homebrew (to get neovim) and VScode on, zero other customization.

Now to hopefully move forward:

> I closed this issue because we cannot reproduce, there is no further information, and no one has this specific issue. It's possible it's a mac or homebrew issue, but I don't own one.

One assumes there are things that can be tested, logs that can be gathered... or do I need to read this statement as a lack of interest in debugging it?

**Comment by theol0403** [comment]
I'll reopen this. Does it still occur? There is code in this extension specifically designed to record keypresses sent after escape, wait until the document is done syncing, and then replay those changes. Strange that it isn't fixed for you. 

I'm thinking of refactoring https://github.com/vscode-neovim/vscode-neovim/pull/992 to notify nvim of changes in real-time, as then there is no need to sync on exit. This might fix your problem.

**Comment by jorhett** [comment]
> I'll reopen this. Does it still occur?

Thank you. I don't know, I had to return to VscodeVim because it impacted my work too much. I'll swap back to this extension and test.

**Comment by theol0403** [comment]
@jorhett try to give #992 a try when you can. It might have some bugs (and dot-repeat is broken for text typed in insert mode), but there should be much less escape lag.

**Comment by jorhett** [comment]
Yes it was completely independent of file size. I could recreate on a brand new workspace with a brand new file at the time.

However, sorry -- I've been meaning to get back to you about this. The problem has disappeared. It's just not happening any more. So whatever you changed, or some other change in VsCode has resolved this. I'm really sorry, I've been telling myself to get back to you for a few weeks now.

**Comment by pencilcheck** [comment]
This is happening to me all the time. I thought I had been crazy and I went away from vscode but now because of typesciprt and work I am coming back to it and feeling dreadful about this slow esc stuff. not sure why no one here can reproduce when all you need to do is to type really fast on big moderately big projects
